plaid/src/apis/npm/npm_web_client.rs

```diff
@@ -0,0 +1,833 @@
+        let response = self
+            .client
+            .get(format!(
+                "{}/settings/{}/members",
+                NPMJS_COM_URL, self.config.npm_scope
+            ))
+            .query(&[("selectedTab", "tfa_disabled")])
+            .header("X-Spiferack", "1")
+            .send()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToRetrieveUserList))?
+            .json::<Value>()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToRetrieveUserList))?;
+
+        // Get total number of users that we will need to retrieve
+        let total_users: u16 = serde_json::from_value(
+            response
+                .get("memberCounts")
+                .ok_or(ApiError::NpmError(NpmError::FailedToRetrieveUserList))?
+                .get("tfa_disabled")
+                .ok_or(ApiError::NpmError(NpmError::FailedToRetrieveUserList))?
+                .clone(),
+        )
+        .map_err(|_| ApiError::NpmError(NpmError::FailedToRetrieveUserList))?;
+
+        let users = self
+            .get_paginated_data_from_npm_website::<NpmUser>(
+                &format!(
+                    "{}/settings/{}/members?selectedTab=tfa_disabled",
+                    NPMJS_COM_URL, self.config.npm_scope
+                ),
+                total_users,
+            )
+            .await?;
+
+        serde_json::to_string(&users)
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToRetrieveUserList))
+    }
+
+    /// Retrieve a list of packages for which a given team has a given permission
+    /// (useful to spot misconfigured package permissions).
+    pub async fn list_packages_with_team_permission(
+        &self,
+        params: &str,
+        module: &str,
+    ) -> Result<String, ApiError> {
+        let params: ListPackagesWithTeamPermissionParams =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+        let team = self.validate_npm_team_name(&params.team)?;
+
+        info!(
+            "Listing all packages for which team [{}] has [{}] access on behalf of [{module}]",
+            team,
+            params.permission.to_string()
+        );
+        self.login()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::LoginFlowError))?;
+
+        let response: Value = self
+            .client
+            .get(format!(
+                "{}/settings/{}/teams/team/{}/access",
+                NPMJS_COM_URL, self.config.npm_scope, team
+            ))
+            .header("X-Spiferack", "1")
+            .send()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToRetrievePackages))?
+            .json()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToRetrievePackages))?;
+
+        // Get total number of packages that we will need to retrieve
+        let total_packages: u16 = serde_json::from_value(
```

Comment:
Interesting choice of `u16`. Is that a limit from NPM? If so where is that documented?

Reply:
Nice catch: changed to `u64`